pkg/reconciler/managed/reconciler.go

```diff
@@ -972,18 +1007,29 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (resu
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	// deep copy the managed resource now that we've called Observe() and have
+	// not performed any external operations - we can use this as the
+	// pre-operation managed resource state in the change logs later
+	//nolint:forcetypeassert // managed.DeepCopyObject() will always be a resource.Managed.
+	managedPreOp := managed.DeepCopyObject().(resource.Managed)
+
 	if meta.WasDeleted(managed) {
 		log = log.WithValues("deletion-timestamp", managed.GetDeletionTimestamp())
 
 		if observation.ResourceExists && policy.ShouldDelete() {
-			if err := external.Delete(externalCtx, managed); err != nil {
+			deletion, err := external.Delete(externalCtx, managed)
+			if err != nil {
 				// We'll hit this condition if we can't delete our external
 				// resource, for example if our provider credentials don't have
 				// access to delete it. If this is the first time we encounter
 				// this issue we'll be requeued implicitly when we update our
 				// status with the new error condition. If not, we want requeue
 				// explicitly, which will trigger backoff.
 				log.Debug("Cannot delete external resource", "error", err)
+				if err := r.changeLogger.RecordChangeLog(ctx, managedPreOp, v1alpha1.OperationType_OPERATION_TYPE_DELETE, err, deletion.AdditionalDetails); err != nil {
+					log.Debug(errRecordChangeLog, "error", err)
+					record.Event(managed, event.Warning(reasonCannotSendChangeLog, err))
```

Comment:
I could pretty easily be convinced that we should just log this, and not create an event.

Change logging not working does seem like more of an operator issue than something an MR author would be concerned with. So maybe we don't emit an event, but log the error at info level?

Reply:
Sounds good, we are no longer recording events and we are logging at info level in https://github.com/crossplane/crossplane-runtime/pull/754/commits/f7fb3a733a63a95ba0d4eb5f5555105d247dabd8